text/0000-fn-lifetime.md

```diff
@@ -0,0 +1,86 @@
+- Feature Name: fn lifetime
+- Start Date: 2017-01-02
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add a `'fn` lifetime that is bound to the scope of the body of the current
+innermost function or closure.
+
+# Motivation
+[motivation]: #motivation
+
+Doing this will enable us to declare that some values live longer than their
+containing scope, allowing us to "allocate" recursive data structures (e.g.
+linked lists) on the stack, as in:
+
+```rust
+struct Node<T> {
+    data: T,
+    next: Option<&'fn Node>,
```

Comment:
@eddyb are you treating this as a item defined in a function body (concrete `'fn` lifetime), or as some late-bound placeholder? Certainly the latter in untenable, but the former could possibly work.